compiler/rustc_borrowck/src/diagnostics/region_name.rs

```diff
@@ -912,7 +913,8 @@ impl<'tcx> MirBorrowckCtxt<'_, 'tcx> {
         };
 
         let tcx = self.infcx.tcx;
-        let region_parent = tcx.parent(region.def_id);
+        let region_def = tcx.generics_of(self.mir_def_id()).region_param(region, tcx).def_id;
+        let region_parent = tcx.parent(region_def);
```

Comment:
🤔 

Is this ever not `mir_def_id`?

Actually nvm wait it can sometimes be the parent impl/trait of a method I guess.

Reply:
Yeah there are a few cases like this where it feels silly to be doing this "get the generics, then get the param then get the parent" dance, but it does matter sometimes.... I don't think we have tests for all such scenarios which is a little scary